### DISCUSSION TITLE
How to create a custom command

### DISCUSSION THREAD

**Discussion by Sadashii** [outline/rich-markdown-editor#618] [Q&A]
Hi, I was looking for a way with which I could create custom command for the editor

**Comment by tommoor** [comment]
What kind of command? Everything except UI can be injected through the `extensions` prop. Existing inbuilt extensions as examples:
https://github.com/outline/rich-markdown-editor/tree/main/src/plugins

**Reply by Sadashii** [reply]
I was hoping for a command that updates a state

**Reply by Sadashii** [reply]
I wanted this to like, /latex, sets a state to true, person inputs some latex, clicks enter and the latex gets appended to the editor

**Reply by Sadashii** [reply]
Although while trying to figure this out, I was messing around the source code and I found out that there are two files /dist/nodes/Math and /dist/nodes/MathDisplay which have code to integrate prose mirror math into the editor but they were never used anywhere and that has me very curious as well

**Reply by tommoor** [reply]
Yes, there is a branch here that was never finished:
https://github.com/outline/rich-markdown-editor/pull/434

The plugin it uses does not support collaborative editing so the work was indefinitely paused.

**Comment by Sadashii** [comment]
Based on use purposes, would it be feasible of the init method could have a "useMath" prop but with the warning that it would disable collaboration, in that case people could prioritize what's more important while having support for maths alongside, for example, in my case as well, collaboration isn't needed but math is

**Reply by tommoor** [reply]
Yes, anyone is welcome to finish that branch, however I'm not planning on looking at it in the near future as collaboration is important for our use case at Outline.

**Reply by Sadashii** [reply]
This topic was started (about how to integrate prose mirror math in outline) over at their reporsity which did get attention by benrbay saying how somewhere around late January he might work at how he could enable collaborative editing (or more specifically to have prosemirror math working with rich-markdown-editor with collaboration feature) 

https://github.com/benrbray/prosemirror-math/issues/32
